boards: arm: nordic nrf91* nrf53* nrf52* nrf51*: Remove scratch areas

nordicjm opened the pull request:
The scratch partition is not needed since MCUboot now operates in
    swap using move mode instead of swap with scratch, as a result, the
    main partitions on Nordic nRF51, nRF52, nRF53 and nRF91 boards can be
    expanded to help in fitting large applications to them.

nordicjm committed:
boards: arm: nordic nrf91* nrf53* nrf52* nrf51*: Remove scratch areas

The scratch partition is not needed since MCUboot now operates in
swap using move mode instead of swap with scratch, as a result, the
main partitions on Nordic nRF51, nRF52, nRF53 and nRF91 boards can be
expanded to help in fitting large applications to them.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic reviewed (changes requested):
It is still possible to configure mcuboot with `CONFIG_BOOT_SWAP_USING_SCRATCH=y`, which will make compilation fail.

nordicjm commented:
> It is still possible to configure mcuboot with CONFIG_BOOT_SWAP_USING_SCRATCH=y, which will make compilation fail.

I see no reason why anyone would do that given that swap using scratch is known broken and upstream MCUboot recommends avoiding it.

nordicjm committed:
tests: subsys: settings: Remove nrf52 scratch

Removes deleting the nRF52840 scratch partition from tests as
this has now been removed.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: subsys: dfu: Remove nrf52 scratch

Removes deleting the nRF52840 scratch partition from tests as
this has now been removed.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

carlescufi reviewed (changes requested):
Two requests:

- Add a release note
- Add somewhere (mcuboot or zephyr) documentation on how to add a scratch partition

nordicjm committed:
doc: release: 3.3: Add note on Nordic board scratch changes

Adds a note on the removal of scratch partitions from Nordic
development boards.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
Added instructions for scratch mode setup here: https://github.com/mcu-tools/mcuboot/pull/1550
